src/color.cpp

```diff
@@ -775,200 +431,51 @@ nc_color rand_color()
-    {"black_yellow"    , c_black_yellow},
-    {"dkgray_yellow"   , c_dkgray_yellow},
-    {"ltgray_yellow"   , c_ltgray_yellow},
-    {"white_yellow"    , c_white_yellow},
-    {"red_yellow"      , c_red_yellow},
-    {"ltred_yellow"    , c_ltred_yellow},
-    {"green_yellow"    , c_green_yellow},
-    {"ltgreen_yellow"  , c_ltgreen_yellow},
-    {"brown_yellow"    , c_brown_yellow},
-    {"yellow_yellow"   , c_yellow_yellow},
-    {"blue_yellow"     , c_blue_yellow},
-    {"ltblue_yellow"   , c_ltblue_yellow},
-    {"magenta_yellow"  , c_magenta_yellow},
-    {"pink_yellow"     , c_pink_yellow},
-    {"cyan_yellow"     , c_cyan_yellow},
-    {"ltcyan_yellow"   , c_ltcyan_yellow},
-    {"black_magenta"   , c_black_magenta},
-    {"dkgray_magenta"  , c_dkgray_magenta},
-    {"ltgray_magenta"  , c_ltgray_magenta},
-    {"white_magenta"   , c_white_magenta},
-    {"red_magenta"     , c_red_magenta},
-    {"ltred_magenta"   , c_ltred_magenta},
-    {"green_magenta"   , c_green_magenta},
-    {"ltgreen_magenta" , c_ltgreen_magenta},
-    {"brown_magenta"   , c_brown_magenta},
-    {"yellow_magenta"  , c_yellow_magenta},
-    {"blue_magenta"    , c_blue_magenta},
-    {"ltblue_magenta"  , c_ltblue_magenta},
-    {"magenta_magenta" , c_magenta_magenta},
-    {"pink_magenta"    , c_pink_magenta},
-    {"cyan_magenta"    , c_cyan_magenta},
-    {"ltcyan_magenta"  , c_ltcyan_magenta},
-    {"black_cyan"      , c_black_cyan},
-    {"dkgray_cyan"     , c_dkgray_cyan},
-    {"ltgray_cyan"     , c_ltgray_cyan},
-    {"white_cyan"      , c_white_cyan},
-    {"red_cyan"        , c_red_cyan},
-    {"ltred_cyan"      , c_ltred_cyan},
-    {"green_cyan"      , c_green_cyan},
-    {"ltgreen_cyan"    , c_ltgreen_cyan},
-    {"brown_cyan"      , c_brown_cyan},
-    {"yellow_cyan"     , c_yellow_cyan},
-    {"blue_cyan"       , c_blue_cyan},
-    {"ltblue_cyan"     , c_ltblue_cyan},
-    {"magenta_cyan"    , c_magenta_cyan},
-    {"pink_cyan"       , c_pink_cyan},
-    {"cyan_cyan"       , c_cyan_cyan},
-    {"ltcyan_cyan"     , c_ltcyan_cyan}
-};
-
 /**
- * Given the name of a color, returns the nc_color constant that matches. If
- * no match is found, c_white is returned.
+ * Given the name of a color, returns the nc_color value that matches. If
+ * no match is found, c_unset is returned.
+ * Special cases:
+ * {"black"           , c_black}, // missing default prefix c_
+ * {"<c|h|i>_black"   , h_black}, // has prefix c_ or h_ or i_
+ * {"dark_gray_red"   , c_dkgray_red}, // dark instead of dk
+ * {"light_blue_red"  , c_ltblue_red}, // light instead of lt
  * @param new_color The color to get, as a std::string.
  * @return The nc_color constant that matches the input.
  */
-nc_color color_from_string(std::string const &new_color)
+nc_color color_from_string(const std::string &color)
 {
-    auto const it = colormap.find(new_color);
-    if (it != std::end(colormap)) {
-        return it->second;
+    std::string new_color = color;
```

Comment:
In this case it's used later in the debugmsg(). Changed it in the other functions though.